### PR TITLE
Add test case for getActiveParticipationInvitation with multiple part…

### LINKED ISSUE #482: Why am I getting all invitations, incl. the ones not assigned to me?
I have made a family study with a father and a mother, and made a deployment with assignments to users like this:

image

So - Jakob is the "Mother" and Aamir is the "Father".

However, when I authenticate to CAWS as "Jakob" (jakob@bardram.net), I get the following invitations from the ParticipationService:

```json
[
    {
        "participation": {
            "studyDeploymentId": "58ce18ff-40e3-4cd0-9fba-fc893d784891",
            "participantId": "264bf3dc-b6c8-4252-b44e-2c5e00262ce9",
            "assignedRoles": {
                "__type": "dk.cachet.carp.common.application.users.AssignedTo.Roles",
                "roleNames": [
                    "Mother"
                ]
            }
        },
        "invitation": {
            "name": "Family Study w. Participant Data",
            "applicationData": "d6d047d5-4cfd-479b-87ac-99f4b5fee172"
        },
        "assignedDevices": [
            {
                "device": {
                    "__type": "dk.cachet.carp.common.application.devices.Smartphone",
                    "roleName": "Mother's Phone",
                    "isPrimaryDevice": true
                }
            }
        ]
    },
    {
        "participation": {
            "studyDeploymentId": "58ce18ff-40e3-4cd0-9fba-fc893d784891",
            "participantId": "5b486d25-b75e-4116-9a4f-eea8105ceea0",
            "assignedRoles": {
                "__type": "dk.cachet.carp.common.application.users.AssignedTo.Roles",
                "roleNames": [
                    "Father"
                ]
            }
        },
        "invitation": {
            "name": "Family Study w. Participant Data",
            "applicationData": "d6d047d5-4cfd-479b-87ac-99f4b5fee172"
        },
        "assignedDevices": [
            {
                "device": {
                    "__type": "dk.cachet.carp.common.application.devices.Smartphone",
                    "roleName": "Father's Phone",
                    "isPrimaryDevice": true
                }
            }
        ]
    },
]
```

So - even though I autheticate as the "Mother" I get **BOTH** invitations????

This entails, that I (as a participant) could choose to be the Father, even though the researcher has assigned me the Mother role......

### PR DESCRIPTION
**Pull request by yuanchen233**
…icipation

### CODE REVIEW

File: carp.deployments.core/src/commonTest/kotlin/dk/cachet/carp/deployments/application/ParticipationServiceTest.kt [hunk 1]
```diff
@@ -76,6 +76,42 @@ interface ParticipationServiceTest
         assertEquals( setOf( expectedAssignedDevice ), retrievedInvitation.assignedDevices )
     }
 
+    @Test
+    fun getActiveParticipationInvitations_with_multiple_participation_succeeds() = runTest {
+        val (participationService, deploymentService, accountService) = createSUT()
+        val protocol = createSinglePrimaryDeviceProtocol()
```
Comment (Whathecode) [inline carp.deployments.core/src/commonTest/kotlin/dk/cachet/carp/deployments/application/ParticipationServiceTest.kt]: It's best for integration tests to try to reflect real-world scenarios. Especially, given that these tests are also used to output _example JSON requests_ which are included in releases (generated through `OutputParticipationServiceTestRequests`, but added manually), and these output JSON requests in turn are used for regression testing when API versions are bumped to test API migrations (`BackwardsCompatibilityTest`).

It's not very common/expected (though possible) to have a study protocol with a single device, which is assigned to both a "father" and "mother". Instead, we could make this test more representative of a multi-device scenario, which in addition also increases code coverage on top of what is already tested in the test above.

**Review by Whathecode — CHANGES_REQUESTED** [review]

File: carp.deployments.core/src/commonTest/kotlin/dk/cachet/carp/deployments/application/ParticipationServiceTest.kt [hunk 2]
```diff
@@ -76,6 +76,42 @@ interface ParticipationServiceTest
...
+        val identities = listOf(
+            AccountIdentity.fromEmailAddress("father@test.com"),
+            AccountIdentity.fromEmailAddress("mother@test.com")
+        )
+        val invitation = StudyInvitation("Test study", "description", "Custom data")
+        val participantInvitations = identities.map { identity ->
+            ParticipantInvitation(
+                participantId = UUID.randomUUID(),
+                AssignedTo.All,
```
Comment (Whathecode) [inline carp.deployments.core/src/commonTest/kotlin/dk/cachet/carp/deployments/application/ParticipationServiceTest.kt]: This means that both accounts will be assigned to _all_ roles (both "father" and "mother"), which while possible, is misleading and makes this example less sensical.

**Review by Whathecode — DISMISSED** [review]
To keep the test focused, I added a `createTwoDevicesAndRolesProtocol()` test helper, which is likely to also be useful in future test cases.

I added additional assertions to verify _both_ accounts, as well as the devices they were assigned.

Lastly, I fixed to code styles to follow the non-idiomatic style of this project. ;) I think the static checker only checks for it in main sources, or it failed here. Either way: I'll gladly polish minor stuff like that in any PR you send along. :)

**Review by Whathecode — APPROVED** [review]
I added an exception for test sources for detekt flagging destructuring operations > 3 fields.

**Comment by yuanchen233** [comment]
Thank you for the explanation and commentnt.

> To keep the test focused, I added a `createTwoDevicesAndRolesProtocol()` test helper, which is likely to also be useful in future test cases.
> 
I thought about re-using `createComplexProtocol` and expose Roles, but this information is not used by other test cases. This multi role protocol should also be relevant when adding more `InputDataType`.

I'll be adopting to the decision making logic (and maybe code style here)

**Pull request merged**